app/models/adventures/adventure_mission.rb

```diff
@@ -6,20 +6,30 @@ class AdventureMission < ApplicationRecord
   translates :title, :description
   globalize_accessors locales: [:en, :sv], attributes: [:title, :description]
 
-  attr_accessor :finished, :points
+  attr_accessor :finished, :points, :accepted, :require_acceptance
 
   validates :title_sv, presence: true
   validates :index, numericality: { greater_than_or_equal: 0 }, presence: true
   validates :max_points, numericality: { greater_than: 0 }, presence: true
+  validates :require_acceptance, null: false
 
   scope :by_group, ->(group) { where(group: group) }
 
+  # Mission is regarded finished when a Mentor marked it as such
+  # (a corresponding AdventureMissionGroup exists)
   def finished?(group)
     groups.include?(group)
   end
 
+  # Mission is regarded accepted when an admin has accepted it
+  # (pending == false in the corresponding AdventureMissionGroup)
+  def accepted?(group)
+    finished?(group) && adventure_mission_groups.by_group(group).first.pending == false
+  end
+
```

Comment:
I think this logic with the states `finished` and `accepted` is a bit confusing. Have you thought about instead having the states `finished` and `pending`? Changing this may however require quite a lot of work, but I don't know.

Reply:
I actually thought it would be clearer that way, however I understand that since they sound very similar. I guess :accepted is only an inverse of :pending for the existing objects. Most confusing is probably calling a mission finished before acceptance.

Reply:
'finished' would probably better be explained by something like 'performed'